### PR TITLE
feat(FX-4772): Inline price control with Create alert modal

### PR DESCRIPTION
**Pull request by nickskalkin**
The type of this PR is: **Feat**

This PR solves [FX-4772]

### Description

[FX-4772]: https://artsyproduct.atlassian.net/browse/FX-4772?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

- [Review app](https://alerts-price-range.artsy.net/)

### Demo

https://github.com/artsy/force/assets/3934579/8a480f39-b13b-4916-932c-b318eefd33cf

### CODE REVIEW

File: src/Apps/Settings/Routes/SavedSearchAlerts/Components/SavedSearchAlertEditForm.tsx [hunk 1]
```diff
@@ -172,9 +185,18 @@ const SavedSearchAlertEditForm: React.FC<SavedSearchAlertEditFormProps> = ({
...
-                    onDeletePress={removePill}
+                    onDeletePress={handleRemovePillPress}
                   />
                 </Flex>
+
+                <Separator my={1} />
+
+                <PriceRangeFilter
+                  onChange={updatePriceRange}
+                  priceRange={criteria.priceRange ?? DEFAULT_PRICE_RANGE}
```
Comment (dimatretyak) [inline src/Apps/Settings/Routes/SavedSearchAlerts/Components/SavedSearchAlertEditForm.tsx:196]: **suggestion**: keep this logic inside the `PriceRangeFilter`. Something like this

```javascript
export const PriceRangeFilter = () => {
  const { criteria, addCriteriaValue } = useSavedSearchAlertContext()

  const handlePriceRangeUpdate = (updatedRange: CustomRange) => {
    addCriteriaValue("priceRange", updatedRange.join("-"))
  }

  return (
    <Expandable
      label="Set price range you are interested in"
      expanded
      borderColor="white100"
    >
      <PriceRange
        priceRange={criteria.priceRange ?? DEFAULT_PRICE_RANGE}
        onPriceRangeUpdate={handlePriceRangeUpdate}
      />
    </Expandable>
  )
}
```

File: src/Components/PriceRange/PriceRange.tsx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    </>
+  )
+}
+
+const isAllBarsEmpty = (bars: HistogramBarEntity[]) => {
+  return bars.every(bar => bar.count === 0)
+}
+
+const shouldDisplayHistogram = (bars?: HistogramBarEntity[]) => {
+  if (!bars) return false
```
Comment (dimatretyak) [inline src/Components/PriceRange/PriceRange.tsx:140]: I think `bars` will always be `array` and there won't be a situation where bars is `undefined`

File: src/Components/PriceRange/PriceRange.tsx [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+      {shouldDisplayHistogram(bars) ? (
+        <Histogram
+          bars={bars!}
+          selectedRange={[sliderRange[0], sliderRange[1]]}
+          data-testid="PriceFilterHistogram"
+        />
+      ) : null}
+
+      <Spacer y={shouldDisplayHistogram(bars) ? 2 : 4} />
```
Comment (dimatretyak) [inline src/Components/PriceRange/PriceRange.tsx:111]: **question**: do we really want use different spaces?

Comment (nickskalkin) [reply src/Components/PriceRange/PriceRange.tsx:111]: As for me, it looks way better with bigger Spacer when histogram is absent.

With bigger spacing:
Screenshot 2023-06-01 at 15 10 48

Without:
Screenshot 2023-06-01 at 15 10 59

Comment (dimatretyak) [reply src/Components/PriceRange/PriceRange.tsx:111]: maybe in this case it is better to always use `<Spacer y={4} />`?

**Review by dimatretyak — DISMISSED** [review]

File: src/Components/SavedSearchAlert/Components/PriceRangeFilter.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+export const PriceRangeFilter: FC = () => {
+  const { criteria, setCriteriaValue } = useSavedSearchAlertContext()
+
+  const handlePriceRangeUpdate = (updatedRange: CustomRange) => {
+    setCriteriaValue("priceRange", updatedRange.join("-"))
+  }
+
+  return (
+    <Expandable
+      label="Set price range you are interested in"
```
Comment (dblandin) [inline src/Components/SavedSearchAlert/Components/PriceRangeFilter.tsx:19]: **suggestion / non-blocking**: Minor but we could consider putting this copy in the locale file. Example [here (copy)](https://github.com/artsy/force/blob/92683c40ea6e84f9e35bbf5fba82c63e0bede278/src/System/i18n/locales/en-US/translation.json#L87) and [here (use)](https://github.com/artsy/force/blob/92683c40ea6e84f9e35bbf5fba82c63e0bede278/src/Apps/Artwork/Components/ArtworkSidebar/ArtworkSidebarCreateArtworkAlert.tsx#L89).

**Review by dblandin — DISMISSED** [review]
Nice work on making this PriceFilter control more portable between these different components!

**Review by anandaroop — APPROVED** [review]
Nice work!

**Pull request merged**